### PR TITLE
Woodcutting

### PR DESCRIPTION
**Pull request by coolbop32**
cleared out all of the everything so would only be 1 commit

### CODE REVIEW

File: src/commands/Minion/chop.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+			name = quantity;
+			quantity = null;
+		}
+
+		const log = Woodcutting.Logs.find(
+			log => stringMatches(log.name, name) || stringMatches(log.name.split(' ')[0], name)
+		);
+
+		if (!log) {
+			throw `Thats not a valid log to mine. Valid ores are ${Woodcutting.Logs.map(
```
Comment (coolbop32) [inline src/commands/Minion/chop.ts]: I missed changing this error message

File: src/tasks/minions/woodcuttingActivity.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+			str += `\n\n${user.minionName}'s Woodcutting level is now ${newLevel}!`;
+		}
+
+		const loot = {
+			[Log.id]: quantity
+		};
+
+		// Roll for pet at 1.5x chance
+		if (Log.petChance && rand(1, Log.petChance * 1.5) < quantity) {
+			loot[WoodcuttingPet!.id] = 1;
```
Comment (gc) [inline src/tasks/minions/woodcuttingActivity.ts]: lets use `itemID('Beaver')` instead of this and Items.get

File: src/commands/Minion/chop.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+		// If the user has a dragon axe & over 61 mining provide 10% speed boost
+		const bank = msg.author.settings.get(UserSettings.Bank);
+		if (msg.author.skillLevel(SkillsEnum.Woodcutting) >= 61) {
+			if (bankHasItem(bank, itemID('Infernal axe'))) {
+				timetoChop = Math.floor(timetoChop * 0.86);
+			} else if (bankHasItem(bank, itemID('Dragon axe'))) {
+				timetoChop = Math.floor(timetoChop * 0.9);
+			}
+		}
```
Comment (gc) [inline src/commands/Minion/chop.ts]: replace this with how mining now does it (dont worry about gilded/3a axe for now)

File: src/commands/Minion/minion.ts [hunk 4]
```diff
@@ -93,6 +93,10 @@ ${Emoji.Smithing} Smithing: ${msg.author.skillLevel(
 			SkillsEnum.Smithing
 		)} (${msg.author.settings.get(UserSettings.Skills.Smithing).toLocaleString()} xp)
 ${Emoji.QuestIcon} QP: ${msg.author.settings.get(UserSettings.QP)}
+		
+${Emoji.Woodcutting} Woodcutting: ${msg.author.skillLevel(
+			SkillsEnum.Woodcutting
+		)} (${msg.author.settings.get(UserSettings.Skills.Woodcutting).toLocaleString()} xp)
```
Comment (gc) [inline src/commands/Minion/minion.ts]: keep QP at the bottom

File: src/commands/Minion/chop.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import {
+	determineScaledLogTime,
+	stringMatches,
+	formatDuration,
+	rand,
+	itemNameFromID
+} from '../../lib/util';
+import { BotCommand } from '../../lib/BotCommand';
```
Comment (gc) [inline src/commands/Minion/chop.ts]: have these 2 imports grouped together with the group below

File: src/commands/Minion/chop.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+			throw `Thats not a valid log to chop. Valid logs are ${Woodcutting.Logs.map(
+				log => log.name
+			).join(', ')}.`;
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Woodcutting) < log.level) {
+			throw `${msg.author.minionName} needs ${log.level} Woodcutting to chop ${log.name}.`;
+		}
+
+		// Calculate the time it takes to mine a single log of this type, at this persons level.
```
Comment (gc) [inline src/commands/Minion/chop.ts]: it says mine

File: src/commands/Minion/chop.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+		let timetoChop = determineScaledLogTime(
+			log!.xp,
+			log.respawnTime,
+			msg.author.skillLevel(SkillsEnum.Woodcutting)
+		);
+
+		// If the user has a dragon axe & over 61 mining provide 10% speed boost
```
Comment (gc) [inline src/commands/Minion/chop.ts]: this comment is wrong

File: src/commands/Minion/chop.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+			name = quantity;
+			quantity = null;
+		}
+
+		const log = Woodcutting.Logs.find(
+			log => stringMatches(log.name, name) || stringMatches(log.name.split(' ')[0], name)
+		);
+
+		if (!log) {
+			throw `Thats not a valid log to chop. Valid logs are ${Woodcutting.Logs.map(
```
Comment (gc) [inline src/commands/Minion/chop.ts]: That's

File: src/lib/skills/woodcutting.ts [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+		id: 1519,
+		name: 'Willow Logs',
+		respawnTime: 0,
+		petChance: 289_286
+	},
+	{
+		level: 35,
+		xp: 85,
+		id: 6333,
+		name: 'teak Logs',
```
Comment (gc) [inline src/lib/skills/woodcutting.ts]: Teak

File: src/lib/skills/woodcutting.ts [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+		id: 3239,
+		name: 'Bark',
+		respawnTime: 0,
+		petChance: 214_367
+	},
+	{
+		level: 50,
+		xp: 125,
+		id: 6332,
+		name: 'Mahongany Logs',
```
Comment (gc) [inline src/lib/skills/woodcutting.ts]: Mahogany

File: src/tasks/minions/woodcuttingActivity.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+			channel.send(str);
+			channel
+				.awaitMessages(mes => mes.author === user && saidYes(mes.content), {
+					time: getUsersPerkTier(user) > 1 ? Time.Minute * 10 : Time.Minute * 2,
+					max: 1
+				})
+				.then(messages => {
+					const response = messages.first();
+
+					if (response) {
```
Comment (gc) [inline src/tasks/minions/woodcuttingActivity.ts]:
```js
if (response) {
    if (response.author.minionIsBusy) return;
```

add this here

Comment (coolbop32) [reply src/tasks/minions/woodcuttingActivity.ts]: do you want that as an additional if response after or replace whats on line 66 with that

Comment (gc) [reply src/tasks/minions/woodcuttingActivity.ts]: just add `if (response.author.minionIsBusy) return;` after the existing one

**Pull request merged**